go/ai/generate.go

```diff
@@ -111,6 +112,29 @@ type ModelOptions struct {
 	Versions []string `json:"versions,omitempty"`
 }
 
+// BackgroundModel is a background action for model operations.
+type BackgroundModel struct {
```

Comment:
```suggestion
type backgroundModel struct {
```

Then we should define an interface called `BackgroundModel` that has the functions specific to this, like `Model`, `Embedder`, etc.